roller crashing when setting up a full-node

### Roller version

v1.9.0-alpha-rc09

### Rollpp type (EVM/WASM) and version

EVM

### OS

ubuntu

### Processor Architecture

amd64

### Others

_No response_

### Steps to reproduce it

roller rollapp setup full-node mode

### Expected result

node running

### Actual result

panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/dymensionxyz/roller/utils/sequencer.GetAllP2pPeers({0xc000cd41c0?, 0x0?}, {{0xc0007d5a41, 0x6}, {0xc000d296c1, 0x30}, {0xc000f017d1, 0x10}, {0xc000d29841, 0x30}, ...})
	/go/src/github.com/dymensionxyz/roller/utils/sequencer/sequencer.go:315 +0x285
github.com/dymensionxyz/roller/cmd/rollapp.Cmd.Cmd.func2(0xc000b52c08, {0x38806c0?, 0x4?, 0x1e5b402?})
	/go/src/github.com/dymensionxyz/roller/cmd/rollapp/setup/setup.go:375 +0xc5f
github.com/spf13/cobra.(*Command).execute(0xc000b52c08, {0x38806c0, 0x0, 0x0})
	/root/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x38369e0)
	/root/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/root/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/dymensionxyz/roller/cmd.Execute()
	/go/src/github.com/dymensionxyz/roller/cmd/root.go:28 +0x1a
main.main()
	/go/src/github.com/dymensionxyz/roller/main.go:6 +0xf

### Log Files

_No response_

### Notes

i guess its caused by missing data in sequencer metadata registration

--- a/utils/sequencer/sequencer.go
+++ b/utils/sequencer/sequencer.go
@@ -311,11 +311,7 @@ func GetAllP2pPeers(raID string, hd consts.HubData) ([]string, error) {
 	var peers []string
 
 	for _, s := range sequencers.Sequencers {
-		if len(sequencers.Sequencers) > 1 {
-			peers = append(peers, s.Metadata.P2PSeeds[0])
-		} else {
-			peers = append(peers, s.Metadata.P2PSeeds...)
-		}
+		peers = append(peers, s.Metadata.P2PSeeds...)
 	}
 
 	return peers, nil

--- a/utils/sequencer/sequencer.go
+++ b/utils/sequencer/sequencer.go
@@ -311,7 +311,9 @@ func GetAllP2pPeers(raID string, hd consts.HubData) ([]string, error) {
 	var peers []string
 
 	for _, s := range sequencers.Sequencers {
-		peers = append(peers, s.Metadata.P2PSeeds...)
+		if s.OptedIn && s.Status == "OPERATING_STATUS_BONDED" {
+			peers = append(peers, s.Metadata.P2PSeeds...)
+		}
 	}
 
 	return peers, nil

--- a/utils/sequencer/types.go
+++ b/utils/sequencer/types.go
@@ -39,6 +39,8 @@ type Info struct {
 	UnbondTime time.Time `protobuf:"bytes,10,opt,name=unbond_time,json=unbondTime,proto3,stdtime"                          json:"unbond_time"`
 	// WhitelistedRelayers is an array of the whitelisted relayer addresses. Addresses are bech32-encoded strings.
 	WhitelistedRelayers []string `protobuf:"bytes,13,rep,name=whitelisted_relayers,json=whitelistedRelayers,proto3"                json:"whitelisted_relayers,omitempty"`
+	// opted in defines whether the sequencer can be selected as proposer
+	OptedIn bool `protobuf:"varint,14,opt,name=opted_in,proto3"                                                       json:"opted_in,omitempty"`
 }
 
 type Metadata struct {

